### DISCUSSION TITLE
How to use the script interface

### DISCUSSION THREAD

**Discussion by JailbreakPapa** [mikke89/RmlUi#576] [Q&A] [answered]
Hello,

I was wondering how I can add another scripting language like lua to hook with RMLUI?

**Comment by mikke89** [comment, marked as answer]
Hey, I'm copying this from the chat, in case others come by here:

Generally, you'll have to create new bindings. That is mostly a matter of replicating the methods of the various classes in the library (like Element), however you do that for your language. Then you need to hook into the ElementDocument::LoadInlineScript and LoadExternalScript, to actually load the scripts. And you probably also want to hook into some of our plugin functionality. The Lua plugin should show some of these things, but let me know if anything is unclear.

**Reply by JailbreakPapa** [reply]
Thanks for the quick response, wasn't able to see your message in gitter since it was throwing a fit with me...
